enginetest/join_planning_tests.go

```diff
@@ -499,6 +499,26 @@ where u in (select * from rec);`,
 				types: []plan.JoinType{},
 				exp:   []sql.Row{{1, 2}},
 			},
+			{
+				q:     "select * from uv where not exists (select * from xy where u = 1)",
+				types: []plan.JoinType{},
+				exp:   []sql.Row{{0, 1}, {2, 2}, {3, 2}},
```

Comment:
(minor) Tests look good, although might be hard to remember exactly what these are testing in the future. Might be worth a comment to help future maintainers.

Out of curiosity... I see that we have double nested `not`s which is great. If we put a `not` on the innermost `where` expression, do we handle that correctly? 

And just a general though for the future... these optimizations seem a little tricky to test, since we should get the same returned result set whether they run or not. Not for this PR of course, but maybe that's another test we consider in the future? Run queries with and without optimizations and validate that we get the same results.

Reply:
Yeah these are pretty far away from being joins, would need to split ORs into table unions to expose individual exist expressions for applyJoin transforms. These might be better as plan tests.

Reply:
I think the second question comes back to a better IR for testing individual rules. Still noodling on that.